internal/pkg/instrumentation/bpf/go.opentelemetry.io/otel/traceglobal/probe.go

```diff
@@ -133,12 +148,55 @@ func New(logger *slog.Logger) probe.Probe {
 					FailureMode: probe.FailureModeIgnore,
 				},
 			},
-			SpecFn: loadBpf,
+			SpecFn:        loadBpf,
+			ProcessRecord: c.decodeEvent,
 		},
 		ProcessFn: processFn,
 	}
 }
 
+type recordKind uint32
+
+const (
+	recordKindTelemetry recordKind = iota
+	recordKindConrol
+)
+
+type converter struct {
+	logger *slog.Logger
+
+	uprobeNewStart   *probe.Uprobe
+	uprobeNewStartMu sync.Mutex
+}
+
+func (c *converter) decodeEvent(record perf.Record) (*event, error) {
+	reader := bytes.NewReader(record.RawSample)
+
+	var kind recordKind
+	err := binary.Read(reader, binary.LittleEndian, &kind)
+	if err != nil {
+		return nil, err
+	}
+
+	var e *event
+	switch kind {
+	case recordKindTelemetry:
+		e := new(event)
```

Comment:
This might be an ignorant question. Why is this required if we have `var e *event` declared?

Reply:
The event pointer is declared above, but the memory has not been allocated. Without the allocation there will be a panic: https://go.dev/play/p/AADS3FvylTR

That said, this is shadowing `e` instead of assigning to it.